pynestml/symbols/predefined_types.py

```diff
@@ -225,6 +226,15 @@ def get_integer_type(cls):
         """
         return cls.name2type[cls.INTEGER_TYPE]
 
+    @classmethod
+    def get_template_type(cls, i):
+        """
+        Returns a new type symbol for argument type templating. The templates are uniquely identified with an integer number `i`.
```

Comment:
I don't understand this integer number part, could you add some additional explanation in the docstring?